### ISSUE TITLE
请问，在检测模型det_db_mbv3_new.pth上微调，为什么越训越差呢？

### ISSUE THREAD

**Issue by LinXin04** [WenmuZhou/PytorchOCR#237]
![image](https://user-images.githubusercontent.com/25193686/172331841-1873f893-1421-4e06-9322-ae84d977b0d7.png)

**Comment by LinXin04** [comment]
from addict import Dict

config = Dict()
config.exp_name = 'DBNet'
config.train_options = {
    # for train
    'resume_from': '/workspace/bin/latin_det_weights/det_db_mbv3_new.pth',  # 继续训练地址
    'third_party_name': '',  # 加载paddle模型可选
    'checkpoint_save_dir': f"/workspace/bin/det_latin_output/{config.exp_name}/checkpoint",  # 模型保存地址，log文件也保存在这里
    'device': 'cuda:0',  # 不建议修改
    'epochs': 10,
    'fine_tune_stage': ['backbone', 'neck', 'head'],
    'print_interval': 1,  # step为单位
    'val_interval': 1,  # epoch为单位
    'ckpt_save_type': 'HighestAcc',  # HighestAcc：只保存最高准确率模型 ；FixedEpochStep：每隔ckpt_save_epoch个epoch保存一个
    'ckpt_save_epoch': 4,  # epoch为单位, 只有ckpt_save_type选择FixedEpochStep时，该参数才有效
}

config.SEED = 927
config.optimizer = {
    'type': 'Adam',
    'lr': 0.001,
    'weight_decay': 1e-4,
}
# backbone设置为swin_transformer时使用
# config.optimizer = {
#     'type': 'AdamW',
#     'lr': 0.0001,
#     'betas': (0.9, 0.999),
#     'weight_decay': 0.05,
# }


config.model = {
    # backbone 可以设置'pretrained': False/True
    'type': "DetModel",
    'backbone': {"type": "MobileNetV3", 'pretrained': True},  # ResNet or MobileNetV3
    #'backbone': {"type": "ResNet", 'layers': 50, 'pretrained': True},  # ResNet or MobileNetV3
    # 'backbone': {"type": "SwinTransformer", 'pretrained': True},#swin_transformer
    # 'backbone': {"type": "ConvNeXt", 'pretrained': True},
    'neck': {"type": 'DB_fpn', 'out_channels': 96},
    'head': {"type": "DBHead"},
    'in_channels': 3,
}

config.loss = {
    'type': 'DBLoss',
    'alpha': 1,
    'beta': 10
}

config.post_process = {
    'type': 'DBPostProcess',
    'thresh': 0.3,  # 二值化输出map的阈值
    'box_thresh': 0.7,  # 低于此阈值的box丢弃
    'unclip_ratio': 1.5  # 扩大框的比例
}

# for dataset
# ##lable文件
### 存在问题，gt中str-->label 是放在loss中还是放在dataloader中
config.dataset = {
    'train': {
        'dataset': {
            'type': 'JsonDataset',
            'file': r'/workspace/bin/latin_results/gt.json',
            'mean': [0.485, 0.456, 0.406],
            'std': [0.229, 0.224, 0.225],
            # db 预处理，不需要修改
            'pre_processes': [{'type': 'IaaAugment', 'args': [{'type': 'Fliplr', 'args': {'p': 0.5}},
                                                              {'type': 'Affine', 'args': {'rotate': [-10, 10]}},
                                                              {'type': 'Resize', 'args': {'size': [0.5, 3]}}]},
                              {'type': 'EastRandomCropData', 'args': {'size': [640, 640], 'max_tries': 50, 'keep_ratio': True}},
                              {'type': 'MakeShrinkMap', 'args': {'shrink_ratio': 0.4, 'min_text_size': 8}},
                              {'type': 'MakeBorderMap', 'args': {'shrink_ratio': 0.4, 'thresh_min': 0.3, 'thresh_max': 0.7}}
                              ],
            'filter_keys': ['img_path', 'img_name', 'text_polys', 'texts', 'ignore_tags', 'shape'],  # 需要从data_dict里过滤掉的key
            'ignore_tags': ['*', '###'],
            'img_mode': 'RGB'
        },
        'loader': {
            'type': 'DataLoader',  # 使用torch dataloader只需要改为 DataLoader
            'batch_size': 16,
            'shuffle': True,
            'num_workers': 1,
            'collate_fn': {
                'type': ''
            }
        }
    },
    'eval': {
        'dataset': {
            'type': 'JsonDataset',
            'file': r'/workspace/bin/latin_test/gt.json',
            'mean': [0.485, 0.456, 0.406],
            'std': [0.229, 0.224, 0.225],
            'pre_processes': [{'type': 'ResizeShortSize', 'args': {'short_size': 736, 'resize_text_polys': False}}],
            'filter_keys': [],  # 需要从data_dict里过滤掉的key
            'ignore_tags': ['*', '###'],
            'img_mode': 'RGB'
        },
        'loader': {
            'type': 'DataLoader',
            'batch_size': 1,  # 必须为1
            'shuffle': False,
            'num_workers': 1,
            'collate_fn': {
                'type': 'DetCollectFN'
            }
        }
    }
}

# 转换为 Dict
for k, v in config.items():
    if isinstance(v, dict):
        config[k] = Dict(v)

**Comment by LinXin04** [comment]
from addict import Dict

config = Dict()
config.exp_name = 'DBNet'
config.train_options = {
    # for train
    'resume_from': '/workspace/bin/latin_det_weights/det_db_mbv3_new.pth',  # 继续训练地址
    'third_party_name': '',  # 加载paddle模型可选
    'checkpoint_save_dir': f"/workspace/bin/det_latin_output/{config.exp_name}/checkpoint",  # 模型保存地址，log文件也保存在这里
    'device': 'cuda:0',  # 不建议修改
    'epochs': 10,
    'fine_tune_stage': ['backbone', 'neck', 'head'],
    'print_interval': 1,  # step为单位
    'val_interval': 1,  # epoch为单位
    'ckpt_save_type': 'HighestAcc',  # HighestAcc：只保存最高准确率模型 ；FixedEpochStep：每隔ckpt_save_epoch个epoch保存一个
    'ckpt_save_epoch': 4,  # epoch为单位, 只有ckpt_save_type选择FixedEpochStep时，该参数才有效
}

config.SEED = 927
config.optimizer = {
    'type': 'Adam',
    'lr': 0.001,
    'weight_decay': 1e-4,
}
config.model = {
    # backbone 可以设置'pretrained': False/True
    'type': "DetModel",
    'backbone': {"type": "MobileNetV3", 'pretrained': True},  # ResNet or MobileNetV3
    #'backbone': {"type": "ResNet", 'layers': 50, 'pretrained': True},  # ResNet or MobileNetV3
    # 'backbone': {"type": "SwinTransformer", 'pretrained': True},#swin_transformer
    # 'backbone': {"type": "ConvNeXt", 'pretrained': True},
    'neck': {"type": 'DB_fpn', 'out_channels': 96},
    'head': {"type": "DBHead"},
    'in_channels': 3,
}

config.loss = {
    'type': 'DBLoss',
    'alpha': 1,
    'beta': 10
}

config.post_process = {
    'type': 'DBPostProcess',
    'thresh': 0.3,  # 二值化输出map的阈值
    'box_thresh': 0.7,  # 低于此阈值的box丢弃
    'unclip_ratio': 1.5  # 扩大框的比例
}

**Comment by novioleo** [comment]
eval和train非同分布，在训练集过拟合了
